sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GlobalAddressResolver.java

```diff
@@ -95,61 +97,63 @@ public GlobalAddressResolver(
     }
 
     @Override
-    public Flux<OpenConnectionResponse> openConnectionsAndInitCaches(String containerLink) {
-        checkArgument(StringUtils.isNotEmpty(containerLink), "Argument 'containerLink' should not be null nor empty");
-
+    public Flux<OpenConnectionResponse> openConnectionsAndInitCaches(ProactiveContainerInitConfig proactiveContainerInitConfig) {
         // Strip the leading "/", which follows the same format for document requests
         // TODO: currently, the cache key used for collectionCache is inconsistent: some are using path with "/", some use path with stripped leading "/",
         // TODO: ideally it should have been consistent across
-        String cacheKey = StringUtils.strip(containerLink, Constants.Properties.PATH_SEPARATOR);
-        return this.collectionCache.resolveByNameAsync(null, cacheKey, null)
-                .flatMapMany(collection -> {
-                    if (collection == null) {
-                        logger.warn("Can not find the collection, no connections will be opened");
-                        return Mono.empty();
-                    }
-
-                    return this.routingMapProvider.tryGetOverlappingRangesAsync(
-                                    null,
-                                    collection.getResourceId(),
-                                    PartitionKeyInternalHelper.FullRange,
-                                    true,
-                                    null)
-                            .map(valueHolder -> {
-
-                                if(valueHolder == null || valueHolder.v == null || valueHolder.v.size() == 0) {
-                                    logger.warn(
-                                            "There is no pkRanges found for collection {}, no connections will be opened",
-                                            collection.getResourceId());
-                                    return new ArrayList<PartitionKeyRangeIdentity>();
-                                }
-
-                                return valueHolder.v
-                                        .stream()
-                                        .map(pkRange -> new PartitionKeyRangeIdentity(collection.getResourceId(), pkRange.getId()))
-                                        .collect(Collectors.toList());
-                            })
-                            .flatMapMany(pkRangeIdentities -> this.openConnectionsAndInitCachesInternal(collection, pkRangeIdentities));
-                });
+        return Flux.fromIterable(proactiveContainerInitConfig.getCosmosContainerIdentities())
```

Comment:
I am wonder whether we should limit the concurrency here as there are meta request limit per account

Reply:
Also +1 - canbe done in separate PR - but please keep track of the work